source/source.go

```diff
@@ -77,6 +77,14 @@ type SupplementalSource interface {
 	DisableByDefault() bool
 }
 
+// EventSource is an interface for a source that can send events
+type EventSource interface {
+	Source
+
+	// SetChannel sets the channel
+	SetChannel(chan struct{}) error
```

Comment:
nit/suggestion: maybe indicate in the name what channel we're setting. E.g. `SetNotifyChannel` or smth, maybe you can come up with a better name. And, maybe it should be **Add**SmthChannel calling the method multiple times will not unwire the old notifier channels.